com.microsoft.mrtk.uxcore/Keyboard/NonNative/NonNativeKeyboard.cs

```diff
@@ -0,0 +1,583 @@
+    ///       To retrieve the input from the Keyboard, subscribe to the textEntered event. Note that
+    ///       tapping 'Close' on the Keyboard will not fire the textEntered event. You must tap 'Enter' to
+    ///       get the textEntered event.
+    /// </remarks>
+    public class NonNativeKeyboard : MonoBehaviour
+    {
+        /// <summary>
+        /// The instance of NonNativeKeyboard in the scene.
+        /// </summary>
+        /// <remarks>
+        /// There can only be one instance of NonNativeKeyboard in a given scene.
+        /// </remarks>
+        public static NonNativeKeyboard Instance { get; private set; }
+
+        /// <summary>
+        /// Layout type enum for the type of keyboard layout to use.  
+        /// Used during keyboard spawning to enable the correct keys based on layout type.
+        /// </summary>
+        public enum LayoutType
+        {
+            Alpha,
+            Symbol,
+            URL,
+            Email,
+        }
+
+        #region Callbacks
+
+        /// <summary>
+        /// Fired when the user submits the text (i.e. when 'Enter' button is pressed and SubmitOnEnter is true).
+        /// </summary>
+        [field: SerializeField, Tooltip("Fired when the user submits the text (i.e. when 'Enter' button is pressed and SubmitOnEnter is true).")]
+        public NonNativeKeyboardTextEvent OnTextSubmit { get; private set; }
+
+        /// <summary>
+        /// Fired every time the text changes.
+        /// </summary>
+        [field: SerializeField, Tooltip("Fired every time the text changes.")]
+        public NonNativeKeyboardTextEvent OnTextUpdate { get; private set; }
+
+        /// <summary>
+        /// Fired every time the close button is pressed.
+        /// </summary>
+        [field: SerializeField, Tooltip("Fired every time the close button is pressed.")]
+        public NonNativeKeyboardTextEvent OnClose { get; private set; }
+
+        /// <summary>
+        /// Fired when the keyboard is shown.
+        /// </summary>
+        [field: SerializeField, Tooltip("Fired when the keyboard is shown.")]
+        public UnityEvent OnShow { get; private set; }
+
+        /// <summary>
+        /// Fired when the shift status is changed.
+        /// </summary>
+        [field: SerializeField, Tooltip("Fired when the shift status is changed.")]
+        public NonNativeKeyboardShiftEvent OnKeyboardShifted { get; private set; }
+
+        /// <summary>
+        /// Fired when any key on the keyboard is pressed.
+        /// </summary>
+        [field: SerializeField, Tooltip("Fired when any key on the keyboard is pressed.")]
+        public NonNativeKeyboardPressEvent OnKeyPressed { get; private set; }
+
+        #endregion Callbacks
+
+        #region Properties
+
+        /// <summary>
+        /// Current text of the keyboard.
+        /// </summary>
+        public string Text
+        {
+            get
+            {
```

Comment:
Arrow functions por favor

` get => text`

Reply:
Was trying to be consistent with the setter below

Reply:
We generally have always gone for arrow functions when possible (so far in MRTK3), even when there's a custom setter, I believe... right?

Reply:
In fact you actually did this already elsewhere in the PR :D

![image](https://user-images.githubusercontent.com/5544935/197955945-3b26debf-fcc4-4af9-a046-8de7222f9707.png)

Reply:
Ah OK, fixed!